base/generator.jl

```diff
@@ -20,6 +20,12 @@ function next(g::Generator, s)
     g.f(v), s2
 end
 
+"""
+    generate(f, c...) -> iterator
+
+Apply `f` to each element of `c`.
+For multiple collection arguments, apply f elementwise.
+"""
 generate(f, c...) = Generator(f, c...)
```

Comment:
This is just #10452 - though "not having to think about types" sounds like you're using the wrong language. You have to think about types all the time with Julia, and unless there are actual examples of a lowercase function API returning different types for different inputs, exporting a new function that's just a trivial wrapper around an existing type is not necessary.